axum-extra/src/routing/mod.rs

```diff
@@ -244,41 +241,35 @@ where
     #[cfg(feature = "typed-routing")]
     fn typed_trace<H, T, P>(self, handler: H) -> Self
     where
-        H: Handler<T, B>,
+        H: Handler<T, S, B>,
         T: FirstElementIs<P> + 'static,
         P: TypedPath,
     {
         self.route(P::PATH, axum::routing::trace(handler))
     }
 
-    fn route_with_tsr<T>(mut self, path: &str, service: T) -> Self
+    fn route_with_tsr(mut self, path: &str, method_router: MethodRouter<S, B>) -> Self
```

Comment:
We need `route_service_with_tsr` now, right?

Reply:
Yes!